packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts

```diff
@@ -39,18 +39,23 @@ export const generateAuthRulesFromRoles = (
   roles: Array<RoleDefinition>,
   fields: Readonly<FieldDefinitionNode[]>,
   hideAllowedFields = false,
-  identityPoolId?: string,
+  hasIdentityPoolId: boolean,
 ): Expression[] => {
   const expressions = [];
   expressions.push(qref(methodCall(ref('ctx.stash.put'), str('hasAuth'), bool(true))), set(ref('authRules'), list([])));
   const fieldNames = fields.map((field) => field.name.value);
   roles.forEach((role) => {
-    expressions.push(convertAuthRoleToVtl(role, fieldNames, hideAllowedFields, identityPoolId));
+    expressions.push(convertAuthRoleToVtl(role, fieldNames, hideAllowedFields, hasIdentityPoolId));
   });
   return expressions;
 };
 
-const convertAuthRoleToVtl = (role: RoleDefinition, fields: string[], hideAllowedFields = false, identityPoolId?: string): Expression => {
+const convertAuthRoleToVtl = (
+  role: RoleDefinition,
+  fields: string[],
+  hideAllowedFields = false,
+  hasIdentityPoolId: boolean
```

Comment:
Shouldn't the last argument be optional or have a default value?

Reply:
Better to keep the required args before ones with default. Keeping `hasIdentityPoolId` as required makes sense after inspecting the call sites.

Reply:
Sounds good, will re-order the args